test/integration/framework/scheduler_operations.go

```diff
@@ -61,12 +60,12 @@ func NewGardenSchedulerTest(ctx context.Context, shootGardenTest *ShootGardenerT
 		return nil, err
 	}
 
-	cloudProfileForShoot := &gardenv1beta1.CloudProfile{}
-	if err := shootGardenTest.GardenClient.Client().Get(ctx, client.ObjectKey{Namespace: "garden", Name: shootGardenTest.Shoot.Spec.Cloud.Profile}, cloudProfileForShoot); err != nil {
+	cloudProfileForShoot := &gardencorev1alpha1.CloudProfile{}
+	if err := shootGardenTest.GardenClient.Client().Get(ctx, client.ObjectKey{Namespace: "garden", Name: shootGardenTest.Shoot.Spec.CloudProfileName}, cloudProfileForShoot); err != nil {
```

Comment:
when using this method within the delete shoot test, this shoot variable is only initialized with its name and namespace `shoot := &gardencorev1alpha1.Shoot{ObjectMeta: metav1.ObjectMeta{Namespace: *projectNamespace, Name: *shootName}}` (https://github.com/gardener/gardener/pull/1474/files#diff-9159c95dca00a986cca6eb4ccc0cc011R71).
Therefore its cloudprofile cannot be fetched. We should check either if the cloudprofile is defined or fetch the shoot from the apiserver before trying to get its cloudprofile.

Reply:
I think you mean in the framework/shoot_operations!   Thanks for catching it.

Reply:
ah yes you're right. wrong file 👍